helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -573,35 +573,41 @@ public void resetPartition(String clusterName, String instanceName, String resou
     // check the instance is alive
     LiveInstance liveInstance = accessor.getProperty(keyBuilder.liveInstance(instanceName));
     if (liveInstance == null) {
-      throw new HelixException(
-          "Can't reset state for " + resourceName + "/" + partitionNames + " on " + instanceName
-              + ", because " + instanceName + " is not alive");
+      // check if the instance has never exist in the cluster
+      String instancePath = PropertyPathBuilder.instance(clusterName, instanceName);
+      String errMessage;
+      if (!_zkClient.exists(instancePath)) {
+        errMessage = String.format(
+            "Can't reset state for %s.%s on %s, because %s has never existed" + " in cluster %s",
+            resourceName, partitionNames, instanceName, instanceName, clusterName);
+      } else {
+        errMessage = String.format("Can't reset state for %s.%s on %s, because %s does not live in"
+                + " cluster %s anymore", resourceName, partitionNames, instanceName, instanceName,
```

Comment:
The check only shows it is not alive for now. "anymore" is not accurate, we can remove it.

Reply:
Please correct me if I am wrong. In my understanding, if the instance has ever lived in the cluster, we would be able to find it under /instance Path but not under the /liveInstance ?